src/ca/wolfram/beta/symath/VariableNode.java

```diff
@@ -0,0 +1,44 @@
+package ca.wolfram.beta.symath;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by Allan Wang on 2017-05-06.
+ */
+public class VariableNode implements MathNode {
+
+    private final String variable;
+    private final Map<String, Double> variableMap;
+
+    VariableNode(String variable, Map<String, Double> variableMap) {
+        this.variable = variable;
+        this.variableMap = variableMap;
+    }
+
+    @Override
+    public NodeType getType() {
+        return NodeType.VARIABLE;
+    }
+
+    @Override
+    public List<MathNode> getChildren() {
+        return null;
+    }
+
+    @Override
+    public boolean isConstant() {
+        return false;
+    }
+
+    @Override
+    public double eval() {
+        return variableMap.containsKey(variable) ? variableMap.get(variable) : 0;
```

Comment:
would it be better to throw something to tell the user that they didn't set the variable?